lib/format_parser.rb

```diff
@@ -30,25 +41,32 @@ def self.parse_http(url)
     parse(cached_io)
   end
 
-  def self.parse(io)
+  def self.parse(io, natures: @natures.to_a, formats: @formats.to_a, returns: :all)
     # If the cache is preconfigured do not apply an extra layer. It is going
     # to be preconfigured when using parse_http.
     io = Care::IOWrapper.new(io) unless io.is_a?(Care::IOWrapper)
 
+    # How many results has the user asked for? Used to determinate whether an array
+    # is returned or not.
+    amount = case returns
+             when :all
+               @parsers.count
+             when :one
+               1
+             else
+               throw ArgumentError.new(":returns does not match any supported mode (:all, :first)")
```

Comment:
Does it expect `:first` or `:one` here? && could you add a note about this to the readme? :)